File: Week1/SingletonPatternExample/src/Logger.java
public class Logger {
    private static Logger instance;
    private Logger(){
        System.out.println("Initialized Logger");
    }
    public static Logger getLogger(){
        if (instance == null) {
            instance = new Logger();
        }
        return instance;
    }
    public void msg(String msg) {
        System.out.println("Log: " + msg);
    }
}

File: Week1_EngineeringConcepts/Code/1_SingletonPatternExample/Logger.cs
using System;
public class Logger
{
    private static Logger instance;
    private Logger()
    {
        Console.WriteLine("Initialized Logger");
    }

    public static Logger GetLogger()
    {
        if (instance == null)
        {
            instance = new Logger();
        }
        return instance;
    }

    public void Msg(string msg)
    {
        Console.WriteLine("Log: " + msg);
    }
}
